launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -52,6 +52,11 @@
   final List<String> pyFiles;
   final Map<String, String> childEnv;
   final Map<String, String> conf;
+  File workingDir;
```

Comment:
It feels weird to add these fields to this class when they're not used by this class at all. Feels like they should be in `SparkLauncher`.